src/hotspot/cpu/ppc/register_ppc.hpp

```diff
@@ -94,7 +96,7 @@ class Register {
   // accessors
   constexpr int encoding() const { assert(is_valid(), "invalid register"); return _encoding; }
   inline VMReg as_VMReg() const;
-  Register successor() const { return Register(encoding() + 1); }
+  Register successor() const { return Register((encoding() + 1) & (number_of_registers - 1)); }
```

Comment:
I don't think we need this. Only s390 has it, but I don't know the purpose of it. I don't think R0 should be the successor of R31.